frontends/web/src/components/forms/button.test.tsx

```diff
@@ -0,0 +1,24 @@
+import { MemoryRouter } from 'react-router-dom';
+import { render, screen } from '@testing-library/react';
+import { ButtonLink, Button } from './button';
+
+describe('components/forms/button', () => {
+  describe('ButtonLink', () => {
+    it('renders as button when disabled with disabled attribute', () => {
+      render(<ButtonLink to="/settings" disabled>A ButtonLink</ButtonLink>);
+      expect(screen.getByRole('button', { name: /A ButtonLink/i })).toHaveAttribute('disabled');
```

Comment:
From reading several articles, `getByRole` is indeed more suitable than `getByTestId` in some scenarios such as this one. 

I mixed the usage in our codebase because I used whatever is easiest to test. This component for example, clearly will only render one single button (or link), so there's no need to use `getByTestId` for example. Moreover, the one "plus" in using getByRole is accessibility, whereas using testid it will find  whatever element there is as long as testid is there (doesn't matter the "role").

"Role" is for example, here an element could be a button or a link - using getByTestId it will not "care" about it (sure we can maybe test its innerHTML or something - but it seems hacky). However, using `getByRole` we wanna assert that there's a component that **identifies itself as.. a button or a link that _also_ has bla bla bla (whatever the next assertion)** . 

Articles on why role > testid:
1. https://derekndavis.com/posts/getbytestid-overused-react-testing-library (this is a pretty good blog)
2. https://dev.to/jacques_blom/don-t-use-getbytestid-32oj

I tbh have no strong opinion on using role vs testid, however I do think it's nice that we can have some sort of confidence that our element indeed renders as whatever it intends to be.. e.g a button, or a link, or a checkbox and so on.